whole-tale/dashboard
# Automatic navigation to appropriate Run tabs

### Problem
Under certain conditions, the user doesn't necessarily know where they should go next.

Fixes #508 
Fixes #507 
Fixes #506 
Fixes #505 

### Approach
Help "guide" the user to where we think they want to go:
* Disable the "Run > Interact" tab when the Tale is not running
* When a user creates a new Tale from the Browse view, they are taken to the "Run > Interact" tab if the user chose to Start the Tale or "Run > Metadata" if they chose not to start it
* When a user clicks "Run" from the Browse view, they are taken to the "Run > Interact" tab for the now launching Tale
* When a user clicks "View" from the Browse view, they are taken to the "Run > Interact" tab if a Tale instance exists, or to "Run > Metadata" tab for a stopped Tale
* When a user clicks "Run" from the Run view, they are taken to the "Run > Interact" tab for the now launching Tale

### How to Test
Prerequisite: At least one stopped Tale, at least one Tale you don't own

1. Checkout and run this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Navigate to the Browse view and choose a stopped Tale
4. Browse->Metadata 1: Create Tale" (without launching) from Create Tale modal
   * You should be brought to the "Run > Metadata" view for this Tale (#508)
   * You should see that the "Run > Interact" tab is disabled (#506)
5. Browse->Metadata 2: Hover over the Tale image and click the "View" button that appears on hover
   * You should be brought to the "Run > Metadata" view for this Tale (#505)
   * You should see that the "Run > Interact" tab is disabled (#506)
6. Browse->Interact 1: "Create Tale And Launch" from Create Tale modal
   * You should be brought to the "Run > Metadata" view for this Tale (#508)
   * You should see that the "Run > Interact" tab is disabled (#506)
7. Browse->Interact 2: Go back to the Browse view and click the "Run" button beneath a stopped Tale
    * You should be brought immediately to the "Run > Interact" view (#507)
    * You should see that the Interact tab is now enabled (#506)
    * You should see the status message(s) indicating that your Tale is now launching
8. Browse->Interact 3: Hover over the now-launching Tale's image and click the "View" button that appears on hover
    * You should be brought immediately to the "Run > Interact" view (#507)
    * You should see that the Interact tab is now enabled (#506)
    * You should see the status message(s) indicating that your Tale is now launching
9. On the "Run > Interact" view, wait for the Tale ot start and then click the "Stop" button to stop it again
    * You should see that the Interact tab is now disabled
10. Run->Interact: Navigate to the "Run > Metadata" view for the now stopped Tale and click "Run"
    * You should be brought immediately to the "Run > Interact" view 
    * You should see that the Interact tab is now enabled (#506)
    * You should see the status message(s) indicating that your Tale is now launching
11. Browse->Interact 4 (w/ COL): Go back to the Browse view again and click the "Run" button beneath a Tale you do not own
    * You should be prompted by a modal asking you to "Copy and Run"
12. Confirm by clicking "Copy and Run"
    * The modal should close
    * You should be brought immediately to the "Run > Interact" view (#507)
    * You should see that the Interact tab is enabled (#506)
    * You should see the status message(s) indicating that your Tale is now launching
